baselines/fedstar/README.md

```diff
@@ -0,0 +1,59 @@
+labels: [Audio Classification, Semi Supervised learning]
+dataset: [Ambient Context, Speech Commands]
+---
+
+# FedStar: Federated Self-training for Semi-supervised Audio Recognition
+
+> Note: If you use this baseline in your work, please remember to cite the original authors of the paper as well as the Flower paper.
+
+****Paper:**** :[dl.acm.org/doi/10.1145/3520128](https://dl.acm.org/doi/10.1145/3520128)
+
+****Authors:**** : Vasileios Tsouvalas, Aaqib Saeed, Tanir Özcelebi
+
+****Abstract:**** : Federated Learning is a distributed machine learning paradigm dealing with decentralized and personal datasets. Since data reside on devices such as smartphones and virtual assistants, labeling is entrusted to the clients or labels are extracted in an automated way. Specifically, in the case of audio data, acquiring semantic annotations can be prohibitively expensive and time-consuming. As a result, an abundance of audio data remains unlabeled and unexploited on users’ devices. Most existing federated learning approaches focus on supervised learning without harnessing the unlabeled data. In this work, we study the problem of semi-supervised learning of audio models via self-training in conjunction with federated learning. We propose FedSTAR to exploit large-scale on-device unlabeled data to improve the generalization of audio recognition models. We further demonstrate that self-supervised pre-trained models can accelerate the training of on-device models, significantly improving convergence within fewer training rounds. We conduct experiments on diverse public audio classification datasets and investigate the performance of our models under varying percentages of labeled and unlabeled data. Notably, we show that with as little as 3% labeled data available, FedSTAR on average can improve the recognition rate by 13.28% compared to the fully supervised federated model.
+
+
+## About this baseline
+
+****What’s implemented:**** : The code is structured in such a way that all experiments for ambient context and speech commands can be derived.
+
+****Datasets:**** : Ambient Context, Speech Commands
+
+****Hardware Setup:**** : These experiments were run on a linux server with 56 CPU threads with 325 GB Ram with A10 GPU in it. Any machine with 8 CPU cores and 16 GB memory or more would be able to run it in a reasonable amount of time. Note: I have install tensorflow with GPU support but by default, the entire experiment runs on CPU-only mode. For cpu you need to replace value of gpus to None present in distribute_gpus function inside clients.py file.
+
+****Contributors:****: Raj Parekh [GitHub](https://github.com/Raj-Parekh24), [Mail](rajparekhwc@gmail.com)
+
+
+## Environment Setup
+# Set python version
+pyenv local 3.10.6
+# Tell poetry to use python 3.10
+poetry env use 3.10.6
+# Now install the environment
+poetry install
+# Start the shell to activate your environment.
+poetry shell
```

Comment:
please put this inside a block of code, not just like plain text. You can see in the readme it just doesn't look so good. (see how `FedProx` (or any other baseline currently under review) does it: https://github.com/adap/flower/tree/main/baselines/fedprox#environment-setup)

Reply:
Thanks for the suggestion, I have transferred it to block of code.